detection-rules/link_google_open_redirect_with_suspicious_indicators.yml

```diff
@@ -0,0 +1,81 @@
+name: "Open Redirect: Google domain with /url path and suspicious indicators"
+description: |
+  This rule examines messages containing image attachments that utilize Google's open redirect (Google.com/url..). 
+  To enhance accuracy and minimize false positives, the rule conducts additional assessments for suspicious indicators, as indicated in the comments.
+type: "rule"
+severity: "medium"
+source: |
+  type.inbound
+  // All attachments are images
+  and length(attachments) > 0
+  and all(attachments, .file_type in ('png', 'jpg', 'bmp'))
+  and sender.email.domain.root_domain not in $org_domains
+  
+  // not a reply
+  and (
+    length(headers.references) == 0
+    or not any(headers.hops, any(.fields, strings.ilike(.name, "In-Reply-To")))
+  )
+  // With a Google Redirect
+  and any(body.links,
+          .href_url.domain.sld == "google"
+          and .href_url.path == "/url"
+          and regex.contains(.href_url.query_params, "hl=.{2}&q=http(s)?://")
+  )
+  and 2 of (
+    // Not a google logo
+    any(attachments,
+        .file_type in ('png', 'jpg', 'bmp')
+        and (
+          any(ml.logo_detect(.).brands, not strings.starts_with(.name, "Google"))
+          or any(ml.logo_detect(beta.message_screenshot()).brands,
+                 not strings.starts_with(.name, "Google")
```

Comment:
does this return true or false if no brands are detected? if unsure, could make the explicit in the logic

Reply:
It returns false.